libpod/image/image.go

```diff
@@ -383,8 +386,11 @@ func (i *Image) UntagImage(tag string) error {
 			newTags = append(newTags, t)
 		}
 	}
+	if err := i.imageruntime.store.SetNames(i.ID(), newTags); err != nil {
+		return err
+	}
 	i.reloadImage()
```

Comment:
To be safe, we may want to have one of these at the beginning of the function too, to grab an accurate list of names to set

Reply:
i thought of that too but you that reread has cost (locks, io) and therefore my feeling was if something modifies the image, it should reload it afterwards.

Reply:
My concern is concurrency - two 'podman tag' launched at basically the same time in different terminals.

Reply:
fixed